Sources/ProjectSpec/SpecValidation.swift

```diff
@@ -194,3 +195,36 @@ extension Project {
         }
     }
 }
+
+extension Dictionary where Key == String, Value: Any {
+
+    public func validateWarnings() throws {
+        var errors: [SpecValidationError.ValidationError] = []
```

Comment:
Could we move the contents of this function to `SpecLoader.validateProjectDictionaryWarnings`. This doesn't need to be on every dictionary

Reply:
I moved it into a private extension within `SpecLoader`.